src/v/iceberg/schema_json.cc

```diff
@@ -0,0 +1,80 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "iceberg/schema_json.h"
+
+#include "iceberg/datatypes.h"
+#include "iceberg/datatypes_json.h"
+#include "iceberg/json_utils.h"
+#include "iceberg/schema.h"
+#include "json/document.h"
+
+namespace iceberg {
+
+schema parse_schema(const json::Value& v) {
+    auto type = parse_required_str(v, "type");
+    if (type != "struct") {
+        throw std::invalid_argument(
+          fmt::format("Schema has type '{}' instead of 'struct'", type));
+    }
+    int32_t schema_id = parse_required_i32(v, "schema-id");
```

Comment:
`_required_` here means that it throws if it doesn't find what is expected right?

Reply:
That's right